haystack/components/generators/chat/openai.py

```diff
@@ -422,15 +416,9 @@ async def _handle_async_stream_response(
         chunk_delta: StreamingChunk
 
         async for chunk in chat_completion:  # pylint: disable=not-an-iterable
-            # choices is an empty array for usage_chunk when include_usage is set to True
-            if chunk.usage is not None:
-                chunk_delta = self._convert_usage_chunk_to_streaming_chunk(chunk)
-
-            else:
-                assert len(chunk.choices) == 1, "Streaming responses should have only one choice."
-                chunk_delta = self._convert_chat_completion_chunk_to_streaming_chunk(chunk)
+            assert len(chunk.choices) <= 1, "Streaming responses should have at most one choice."
+            chunk_delta = self._convert_chat_completion_chunk_to_streaming_chunk(chunk)
```

Comment:
Due to the update of `_convert_chat_completion_chunk_to_streaming_chunk` we no longer need this if check.